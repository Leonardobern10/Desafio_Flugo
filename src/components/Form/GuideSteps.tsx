import {Step, StepLabel, Stepper, Typography} from '@mui/material';
import type {ReactElement} from 'react';
import type {GuideStepsProps} from '../../types/GuideStepsProps';

export default function GuideSteps({
     activeStep,
     steps,
}: GuideStepsProps): ReactElement {
     return (
          <Stepper
               activeStep={activeStep}
               orientation="vertical" // ou 'horizontal' se preferir
               connector={null}
               sx={{
                    minWidth: 170,
                    display: 'flex',
                    flexDirection: 'column',
                    rowGap: activeStep === 1 ? 0 : 10,
               }}>
               {steps.map((label) => (
                    <Step key={label.step}>
                         <StepLabel>
                              <Typography fontSize="13px">
                                   {label.step}
                              </Typography>
                         </StepLabel>
                    </Step>
               ))}
          </Stepper>
     );
}
