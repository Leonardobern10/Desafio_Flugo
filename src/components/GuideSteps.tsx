import {Step, StepLabel, Stepper} from '@mui/material';
import type {ReactElement} from 'react';

export default function GuideSteps(props: {
     activeStep: number;
     steps: Array<string>;
}): ReactElement {
     return (
          <Stepper
               activeStep={props.activeStep}
               orientation="vertical" // ou 'horizontal' se preferir
               sx={{
                    minWidth: 200,
                    height: 'fit-content',
               }}>
               {props.steps.map((label) => (
                    <Step key={label}>
                         <StepLabel>{label}</StepLabel>
                    </Step>
               ))}
          </Stepper>
     );
}
