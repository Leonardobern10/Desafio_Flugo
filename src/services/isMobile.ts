import {useMediaQuery} from '@mui/material';
import theme from '../theme';

/**
 * Custom React Hook para detectar se a largura de tela atual
 * pertence à um dispositivo móvel.
 *
 * Utiliza o breakpoint 'md' do tema Material UI para verificar se a tela
 * está em um tamanho igual ou menor que 'md' (geralmente 960px).
 *
 * @returns {boolean} `true` se a tela for considerada mobile (menor ou igual a 'md'), `false` caso contrário.
 *
 *
 * */
export default function useIsMobile(): boolean {
     return useMediaQuery(theme.breakpoints.down('lg'));
}
