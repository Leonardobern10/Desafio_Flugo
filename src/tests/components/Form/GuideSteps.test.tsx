import {render, screen} from '@testing-library/react';
import isMobile from '@services/isMobile';
import {afterEach, describe, expect, it, vi} from 'vitest';
import GuideSteps from '@components/Form/GuideSteps';

vi.mock('@services/isMobile');

describe('GuideSteps', () => {
     const stepsMock = [
          {step: 'Step 1', title: 'Title 1'},
          {step: 'Step 2', title: 'Title 2'},
          {step: 'Step 3', title: 'Title 3'},
     ];

     afterEach(() => {
          vi.clearAllMocks();
     });

     it('deve renderizar todos os passos fornecidos', () => {
          (isMobile as vi.Mock).mockReturnValue(false);

          render(<GuideSteps activeStep={0} steps={stepsMock} />);

          stepsMock.forEach(({step}) => {
               expect(screen.getByText(step)).toBeInTheDocument();
          });
     });

     it('deve aplicar orientação vertical quando não for mobile', () => {
          (isMobile as vi.Mock).mockReturnValue(false);

          const {container} = render(
               <GuideSteps activeStep={1} steps={stepsMock} />
          );
          const stepper = container.querySelector('.MuiStepper-root');

          expect(stepper).toHaveClass('MuiStepper-vertical');
     });

     it('deve aplicar orientação horizontal quando for mobile', () => {
          (isMobile as vi.Mock).mockReturnValue(true);

          const {container} = render(
               <GuideSteps activeStep={1} steps={stepsMock} />
          );
          const stepper = container.querySelector('.MuiStepper-root');

          expect(stepper).not.toHaveClass('MuiStepper-vertical');
          // opcional: verificar se não tem classe MuiStepper-horizontal, se aplicável
     });

     it('deve ajustar o rowGap conforme activeStep via snapshot', () => {
          (isMobile as vi.Mock).mockReturnValue(false);

          const {container, rerender} = render(
               <GuideSteps activeStep={0} steps={stepsMock} />
          );

          expect(container.firstChild).toMatchSnapshot();

          rerender(<GuideSteps activeStep={1} steps={stepsMock} />);

          expect(container.firstChild).toMatchSnapshot();
     });
});
