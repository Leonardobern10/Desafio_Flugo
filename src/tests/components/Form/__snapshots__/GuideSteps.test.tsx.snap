// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GuideSteps > deve ajustar o rowGap conforme activeStep via snapshot 1`] = `
<div
  class="MuiStepper-root MuiStepper-vertical css-17v9wil-MuiStepper-root"
>
  <div
    class="MuiStep-root MuiStep-vertical css-enonti-MuiStep-root"
  >
    <span
      class="MuiStepLabel-root MuiStepLabel-vertical css-122jq7f-MuiStepLabel-root"
    >
      <span
        class="MuiStepLabel-iconContainer Mui-active css-1b2ju2j-MuiStepLabel-iconContainer"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-3hpdci-MuiSvgIcon-root-MuiStepIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="12"
          />
          <text
            class="MuiStepIcon-text css-1g86afo-MuiStepIcon-text"
            dominant-baseline="central"
            text-anchor="middle"
            x="12"
            y="12"
          >
            1
          </text>
        </svg>
      </span>
      <span
        class="MuiStepLabel-labelContainer css-p8xe3-MuiStepLabel-labelContainer"
      >
        <span
          class="MuiStepLabel-label Mui-active css-12eb3rq-MuiStepLabel-label"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1bkd8o1-MuiTypography-root"
          >
            Step 1
          </p>
        </span>
      </span>
    </span>
  </div>
  <div
    class="MuiStep-root MuiStep-vertical css-enonti-MuiStep-root"
  >
    <span
      class="MuiStepLabel-root MuiStepLabel-vertical Mui-disabled css-122jq7f-MuiStepLabel-root"
    >
      <span
        class="MuiStepLabel-iconContainer Mui-disabled css-1b2ju2j-MuiStepLabel-iconContainer"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-3hpdci-MuiSvgIcon-root-MuiStepIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="12"
          />
          <text
            class="MuiStepIcon-text css-1g86afo-MuiStepIcon-text"
            dominant-baseline="central"
            text-anchor="middle"
            x="12"
            y="12"
          >
            2
          </text>
        </svg>
      </span>
      <span
        class="MuiStepLabel-labelContainer css-p8xe3-MuiStepLabel-labelContainer"
      >
        <span
          class="MuiStepLabel-label Mui-disabled css-12eb3rq-MuiStepLabel-label"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1bkd8o1-MuiTypography-root"
          >
            Step 2
          </p>
        </span>
      </span>
    </span>
  </div>
  <div
    class="MuiStep-root MuiStep-vertical css-enonti-MuiStep-root"
  >
    <span
      class="MuiStepLabel-root MuiStepLabel-vertical Mui-disabled css-122jq7f-MuiStepLabel-root"
    >
      <span
        class="MuiStepLabel-iconContainer Mui-disabled css-1b2ju2j-MuiStepLabel-iconContainer"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-3hpdci-MuiSvgIcon-root-MuiStepIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="12"
          />
          <text
            class="MuiStepIcon-text css-1g86afo-MuiStepIcon-text"
            dominant-baseline="central"
            text-anchor="middle"
            x="12"
            y="12"
          >
            3
          </text>
        </svg>
      </span>
      <span
        class="MuiStepLabel-labelContainer css-p8xe3-MuiStepLabel-labelContainer"
      >
        <span
          class="MuiStepLabel-label Mui-disabled css-12eb3rq-MuiStepLabel-label"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1bkd8o1-MuiTypography-root"
          >
            Step 3
          </p>
        </span>
      </span>
    </span>
  </div>
</div>
`;

exports[`GuideSteps > deve ajustar o rowGap conforme activeStep via snapshot 2`] = `
<div
  class="MuiStepper-root MuiStepper-vertical css-1ovhxo9-MuiStepper-root"
>
  <div
    class="MuiStep-root MuiStep-vertical Mui-completed css-enonti-MuiStep-root"
  >
    <span
      class="MuiStepLabel-root MuiStepLabel-vertical css-122jq7f-MuiStepLabel-root"
    >
      <span
        class="MuiStepLabel-iconContainer Mui-completed css-1b2ju2j-MuiStepLabel-iconContainer"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-completed css-3hpdci-MuiSvgIcon-root-MuiStepIcon-root"
          data-testid="CheckCircleIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
          />
        </svg>
      </span>
      <span
        class="MuiStepLabel-labelContainer css-p8xe3-MuiStepLabel-labelContainer"
      >
        <span
          class="MuiStepLabel-label Mui-completed css-12eb3rq-MuiStepLabel-label"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1bkd8o1-MuiTypography-root"
          >
            Step 1
          </p>
        </span>
      </span>
    </span>
  </div>
  <div
    class="MuiStep-root MuiStep-vertical css-enonti-MuiStep-root"
  >
    <span
      class="MuiStepLabel-root MuiStepLabel-vertical css-122jq7f-MuiStepLabel-root"
    >
      <span
        class="MuiStepLabel-iconContainer Mui-active css-1b2ju2j-MuiStepLabel-iconContainer"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-3hpdci-MuiSvgIcon-root-MuiStepIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="12"
          />
          <text
            class="MuiStepIcon-text css-1g86afo-MuiStepIcon-text"
            dominant-baseline="central"
            text-anchor="middle"
            x="12"
            y="12"
          >
            2
          </text>
        </svg>
      </span>
      <span
        class="MuiStepLabel-labelContainer css-p8xe3-MuiStepLabel-labelContainer"
      >
        <span
          class="MuiStepLabel-label Mui-active css-12eb3rq-MuiStepLabel-label"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1bkd8o1-MuiTypography-root"
          >
            Step 2
          </p>
        </span>
      </span>
    </span>
  </div>
  <div
    class="MuiStep-root MuiStep-vertical css-enonti-MuiStep-root"
  >
    <span
      class="MuiStepLabel-root MuiStepLabel-vertical Mui-disabled css-122jq7f-MuiStepLabel-root"
    >
      <span
        class="MuiStepLabel-iconContainer Mui-disabled css-1b2ju2j-MuiStepLabel-iconContainer"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-3hpdci-MuiSvgIcon-root-MuiStepIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="12"
          />
          <text
            class="MuiStepIcon-text css-1g86afo-MuiStepIcon-text"
            dominant-baseline="central"
            text-anchor="middle"
            x="12"
            y="12"
          >
            3
          </text>
        </svg>
      </span>
      <span
        class="MuiStepLabel-labelContainer css-p8xe3-MuiStepLabel-labelContainer"
      >
        <span
          class="MuiStepLabel-label Mui-disabled css-12eb3rq-MuiStepLabel-label"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1bkd8o1-MuiTypography-root"
          >
            Step 3
          </p>
        </span>
      </span>
    </span>
  </div>
</div>
`;
